wordpress-mobile/WordPress-Android
# Stats revamp v2: Update Insights to new set of defaults

This PR separates insights into old and new and updates to new through feature announcement try it now action.

Review/ Merge Instructions

1. Make sure [the corresponding FluxC PR is merged to trunk](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2453).
2. Replace FluxC trunk- in build.gradle.
3. Remove Not Ready for Merge label.
4. Merge the PR.


Fixes #16686 

To test:

Test 1:
- Fresh install Jetpack app and Launch
- Navigate to stats and ensure previous default set of cards;  are displayed in insights tab
- Ensure that default set of cards Most Popular, All time, Today, and Followers are displayed

Test 2:
- Enable StatsRevampv2FeatureConfig and relaunch app
- Ensure Feature announcement dialog is shown
- Tap on try it now action, it should take to insights and show new set of defaults that include Views and Visitors, Total Likes, Total Comments, Total Followers, Most Popular and Latest Post Summary cards

Test 3:
- Fresh install Jetpack app and Launch
- Navigate to stats and ensure previous default set of cards are displayed in insights tab
- Go to insights management, add some cards (like This Year, Posting Activity for example) and save and return to insights, make sure the added cards are shown
- Enable StatsRevampv2FeatureConfig and relaunch app
- Ensure Feature announcement dialog is shown
- Tap on try it now action, it should take to insights and show new set of defaults as in Test 2 above at the top while keeping all the cards added below new set of cards.

## Regression Notes
1. Potential unintended areas of impact
Not that aware of

3. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

4. What automated tests I added (or what prevented me from doing so)
Updated existing unit tests

PR submission checklist:
